### PR TITLE
O_Bilynska Java8

### PR DESCRIPTION
**Pull request by Bilunska**

### CODE REVIEW

File: src/hw java8/src/main/java/com/company/hw/User.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    //1
+    void getAvailableBooks() {
+        LocalDate lastDay =
+                date.with(TemporalAdjusters.lastDayOfMonth());
+        DateTimeFormatter format =
+                DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+        String formatedDateTime = lastDay.format(format);
+
+        books.stream().forEach(element -> {
```
Comment (nmelnyk-corevalue) [inline src/hw java8/src/main/java/com/company/hw/User.java]: can we do stream->filter->foreach ... just to avoid IF

Comment (MychkaO) [reply src/hw java8/src/main/java/com/company/hw/User.java]: FIXED

File: src/hw java8/src/main/java/com/company/hw/User.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+        String formatedDateTime = lastDay.format(format);
+        books.stream()
+                .filter(element -> element.available)
+                .forEach(element -> System.out.println(String.format("%s available till %s", element, formatedDateTime)));
+    }
+
+
+    //2
+    void getFilteredByAuthor(String author) {
```
Comment (nmelnyk-corevalue) [inline src/hw java8/src/main/java/com/company/hw/User.java]: getFilteredByAuthor method with returning void -> bad idea
(ListOfBooks) getFilteredBooksByAuthor(String author) 
OR
void filterBooksByAuthor(Str author)

File: src/hw java8/src/main/java/com/company/hw/User.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        if (tempList.isEmpty()) {
+            System.out.println("The last book was brought in " + date);
+            throw new NoSuchBookException("We don't have books in this Genre");
+        }
+
+    }
+
+
+    //6
+    boolean hasNotBooksByAuthor(String author) {
```
Comment (nmelnyk-corevalue) [inline src/hw java8/src/main/java/com/company/hw/User.java]: hasNoBooksByAuthor

File: src/hw java8/src/main/java/com/company/hw/User.java [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        final Optional<Library> optionalBook = books.stream()
+                .filter(element -> element.getAuthor().contains(author))
+                .findFirst();
+        return !optionalBook.isPresent();
+    }
+
+    //7
+    void filteredByGenre() {
```
Comment (nmelnyk-corevalue) [inline src/hw java8/src/main/java/com/company/hw/User.java]: printFilteredBooksByGenre

File: src/hw java8/src/main/java/com/company/hw/books/Comics.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import com.company.hw.interfaces.Artistic;
+
+public class Comics extends Library implements Artistic {
+
+    public Comics(String author, String name, int year, boolean avail){
+        super(author, name, year, avail);
+    }
+
+    @Override
+    public void typing(boolean type) {
```
Comment (nmelnyk-corevalue) [inline src/hw java8/src/main/java/com/company/hw/books/Comics.java]: typing -> type (VERB) -> convention
boolean type -> the same -> should start with is OR has

File: src/hw java8/src/main/java/com/company/hw/books/Encyclopedia.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import com.company.hw.interfaces.NotArtistic;
+
+public class Encyclopedia  extends Library implements NotArtistic {
+
+    public Encyclopedia(String author, String name, int year, boolean avail) {
+        super(author, name, year, avail);
+    }
+
+    @Override
+    public void noTyping(boolean type) {
```
Comment (nmelnyk-corevalue) [inline src/hw java8/src/main/java/com/company/hw/books/Encyclopedia.java]: the same as previous

File: src/hw java8/src/main/java/com/company/hw/books/Novel.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
+package com.company.hw.books;
+
+import com.company.hw.Library;
+import com.company.hw.interfaces.Artistic;
+
+public class Novel extends Library implements Artistic {
+    private int rozdil;
```
Comment (nmelnyk-corevalue) [inline src/hw java8/src/main/java/com/company/hw/books/Novel.java]: maybe rozdil -> chapter?

File: src/hw java8/src/main/java/com/company/hw/User.java (hunk 2 above)
Comment (MychkaO) [reply src/hw java8/src/main/java/com/company/hw/User.java]: Fixed

File: src/hw java8/src/main/java/com/company/hw/User.java (hunk 4 above)
Comment (MychkaO) [reply src/hw java8/src/main/java/com/company/hw/User.java]: Fixed

File: src/hw java8/src/main/java/com/company/hw/books/Comics.java (hunk 5 above)
Comment (MychkaO) [reply src/hw java8/src/main/java/com/company/hw/books/Comics.java]: Fixed

File: src/hw java8/src/main/java/com/company/hw/User.java (hunk 3 above)
Comment (MychkaO) [reply src/hw java8/src/main/java/com/company/hw/User.java]: Fixed

File: src/hw java8/src/main/java/com/company/hw/books/Novel.java (hunk 7 above)
Comment (MychkaO) [reply src/hw java8/src/main/java/com/company/hw/books/Novel.java]: Fixed

File: src/hw java8/src/main/java/com/company/hw/books/Encyclopedia.java (hunk 6 above)
Comment (MychkaO) [reply src/hw java8/src/main/java/com/company/hw/books/Encyclopedia.java]: Fixed

**Comment by olobanok** [comment]
code is not placed in source folder
please rearrange packages so that all your code is placed under src -> main -> java

**Pull request closed without merging**